pkg/services/provisioning/dashboards/types.go

```diff
@@ -73,10 +73,16 @@ func createDashboardJson(data *simplejson.Json, lastModified time.Time, cfg *Das
 	return dash, nil
 }
 
-func mapV0ToDashboardAsConfig(v0 []*DashboardsAsConfigV0) []*DashboardsAsConfig {
+func mapV0ToDashboardAsConfig(v0 []*DashboardsAsConfigV0) ([]*DashboardsAsConfig, error) {
 	var r []*DashboardsAsConfig
+	seen := make(map[string]bool)
 
 	for _, v := range v0 {
+		if _, ok := seen[v.Name]; ok {
+			return nil, fmt.Errorf("name %s is not unique", v.Name)
```

Comment:
Not sure whether this should fail all configs when a duplicate is found and after or if we just should log an error? Leaning towards logging an error.

Reply:
@grafana/backend-platform and/or @sakjur do you have any preferences regarding return and fail early or logging the error and let the provisioning continue?

Reply:
🤔 I don't think I have an opinion on that either way. Fail early is nice if that'd mean the administrator understands that something is wrong faster, but is probably not desirable for other reasons (since the provisioning isn't done only on boot unless I'm mistaken, what would the consequence be of the provisioning stopping here?)

Reply:
Would logging error only fix the problem mentioned in #20421 ?

Reply:
@marefr I haven't tried Grafana provisioning before. How can we show duplicate errors to the user, so they are likely to notice it, if not failing hard? I think I would prefer not failing and informing the user at the end, if possible.

Reply:
@aknuds1 Wouldn't that impact the workflow? (I mean, failing hard before the final run, rather than not failing, informing the user at the end and then adding another thing to the debugging pool)

Reply:
I think this should fail fast. Its invalid configuration and something the administrator have to deal with.

Reply:
@aknuds1 @marefr @sakjur any insights about that? Should I print the error at the end or should it fail?

Reply:
@youshy After thinking about this I think this should fail fast. It's invalid configuration and something the administrator have to deal with.